docs/sql/statements/select.rst

```diff
@@ -304,6 +304,43 @@ each group.
   column from the queried relation rather than an output column name.
 
 
+.. _sql-select-group-by-all:
+
+``GROUP BY ALL``
+''''''''''''''''
+
+``GROUP BY ALL`` is a convenient shorthand syntax that automatically groups by all non-aggregate
+columns in the SELECT list. This eliminates the need to explicitly list all grouping columns.
+
+::
+
+   GROUP BY ALL [HAVING condition]
+
+When using ``GROUP BY ALL``, the following rules apply:
+- All output columns that are not aggregate functions are used as grouping columns
+- Aggregate functions (like COUNT, SUM, AVG etc.) are excluded from the automatic grouping
+- The order of columns in the grouping matches their order in the SELECT list
+
+    For example::
+
+        SELECT department, title, AVG(salary) as avg_salary
+        FROM employees
+        GROUP BY ALL;
+
+    Is equivalent to explicitly listing all non-aggregate columns::
+
+        SELECT department, title, AVG(salary) as avg_salary
+        FROM employees
+        GROUP BY department, title;
+
+.. NOTE::
+
+  The same limitations that apply to regular GROUP BY also apply to GROUP BY ALL:
+  - Grouping can only be applied on indexed fields
```

Comment:
Do we have these limitations listed elsewhere? If so, can you link to that location instead to avoid duplicating this?

The first point is afaik no longer true, so we'll have to update that (but we can do that separate from this PR)

Reply:
No i did not find that anywhere else , do you want me to remove the notes section completely ?

Reply:
@mfussenegger @nomicode @dobe